Library/Homebrew/utils/curl.rb

```diff
@@ -303,6 +303,85 @@ def curl_http_content_headers_and_checksum(url, specs: {}, hash_needed: false, u
     def http_status_ok?(status)
       (100..299).cover?(status.to_i)
     end
+
+    # Separates the output text from `curl` into an array of response heads and
+    # the final response body.
+    # @param output [String] The output text from `curl` containing
+    #   response head(s), body, or both.
+    # @return [Array<Array<Hash>, String>] An array containing an array of
+    #   the response heads and the body output, if found.
```

Comment:
Thinking about this more, would it be better for the return value to simply be a hash with a format like `{ heads: [{...}, ...], body: "..." }`? It seems like `response[:heads]` may be easier to remember compared to `response[0]`. Similarly, `response[:body]` would be more straightforward than `response[1]`.